.github/workflows/codeql-analysis.yml

```diff
@@ -38,34 +39,18 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
```

Comment:
(should) use a SHA like so:

```yaml
      uses: actions/checkout@<sha> # tag=v3
```

note: `v3` is a _branch_ (by convention)--not a tag, despite the above comment.  

if/when we switch over to Renovate, it will manage this for us, keeping the SHAs updated.